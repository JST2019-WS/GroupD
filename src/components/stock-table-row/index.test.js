/* eslint-env node, jest */

import { h } from "preact";
import StockTableRow from "./index";
import {shallow, mount} from 'enzyme'
import stocks from '../../../test/stocks.json'

const stock = stocks[0];

describe('Stock Table Row', () => {
    it('should display the stock', () => {
        const component = shallow(<StockTableRow stock={stock}/>);
        expect(component.html()).toMatchSnapshot()
    });

    it('should render negative/positive numbers with the correct class', () => {
        const component = shallow(<StockTableRow stock={stock}/>);

        const testFormat = (num, node, prop) => {
            stock[prop] = num;
            component.update();
            const format = num >= 0.0 ? 'positiveNumber' : 'negativeNumber';
            expect(node.hasClass(format))
        };

        for(const [childID, prop] of [[3, 'absolute'], [4, 'relative']]) {
            for(const num in [-1.0, 1.0, 0.0]) {
                testFormat(1.0, component.find('td').at(childID), prop)
            }
        }
    });

    it('should render an error on passing a null stock', () => {
        const component = shallow(<StockTableRow stock={null}/>);
        expect(component.html()).toMatchSnapshot('null-stock')
    });

    it('should render an error on passing a empty stock', () => {
        const component = shallow(<StockTableRow stock={{}}/>);
        expect(component.html()).toMatchSnapshot('null-stock')
    });

    it('should render default values', () => {
        const component = shallow(<StockTableRow stock={{name: 'Test'}}/>);
        expect(component.html()).toMatchSnapshot('default-stock')
    });

    it('should propagate click/hover events', () => {
        const onClick = jest.fn();
        const onHover = jest.fn();
        const component = shallow(<StockTableRow stock={stock} onClick={onClick} onHover={onHover}/>);

        component.simulate('click');
        expect(onClick.mock.calls.length).toBe(1); // Only one event should be raised
        expect(onClick.mock.calls[onClick.mock.calls.length-1][1]).toBe('row');

        component.find('a').first().simulate('click');
        expect(onClick.mock.calls.length).toBe(2); // Clicking on a link does not raise a row event
        expect(onClick.mock.calls[onClick.mock.calls.length-1][1]).toBe('link');

        component.simulate('mouseover');
        expect(onHover.mock.calls.length).toBe(1)
    })
});