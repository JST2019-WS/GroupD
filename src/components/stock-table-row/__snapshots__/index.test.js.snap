// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stock Table Row should display the stock 1`] = `"<tr class=\\"stock-table-row\\"><td class=\\"name\\"><a href=\\"https://www.wallstreet-online.de/aktien/deutsche-pfandbriefbank-aktie\\">Deutsche Pfandbriefbank</a></td><td class=\\"category\\"><a href=\\"https://www.wallstreet-online.de/aktien/branche/banken-aktien\\">Banken</a></td><td class=\\"value\\">10,71 EUR</td><td class=\\"absolute negativeNumber\\">-0,26 EUR</td><td class=\\"relative negativeNumber\\">-2,37 %</td><td class=\\"timestamp small\\">04:57:06</td><td class=\\"exchange small\\">LAS</td><td class=\\"revenue small\\">35.729 EUR</td></tr>"`;

exports[`null-stock 1`] = `"<tr><td class=\\"name\\">Error: Could not load stock</td><td class=\\"category\\">-</td><td class=\\"value\\">-</td><td class=\\"absolute\\">-</td><td class=\\"relative\\">-</td><td class=\\"timestamp small\\">-</td><td class=\\"exchange small\\">-</td><td class=\\"revenue small\\">-</td></tr>"`;
